pwiz/data/msdata/IO.cpp

```diff
@@ -2557,6 +2583,13 @@ void write(minimxml::XMLWriter& writer, const ChromatogramList& chromatogramList
         // write the chromatogram
         if (chromatogram->index != i) throw runtime_error("[IO::write(ChromatogramList)] Bad index.");
         chromatogram->index -= skipped;
+
+        if(!chromatogram->additionalChannelInfo.empty())
+        {
+            chromatogram->userParams.push_back(UserParam("ChannelType",
+                                                         chromatogram->additionalChannelInfo));
+        }
+
```

Comment:
If the purpose of additionalChannelInfo is just to create a userParam, it seems simpler to just add this userParam at the point additionalChannelInfo is set.